File: Administrativo/spring-erp-servicio/net/royal/spring/core/servicio/CompanyownerServicio.cs
using System;
using System.Collections.Generic;
using System.Text;

using net.royal.spring.framework.web.servicio;
using net.royal.spring.core.dominio;
using net.royal.spring.framework.core.dominio.dto;

namespace net.royal.spring.core.servicio
{

public interface CompanyownerServicio : GenericoServicio {
        List<Companyowner> listarTodos();
        string obtenerNombre(string companyowner);
        List<DtoTabla> listarActivos();
        List<DtoTabla> listarCompaniasPorSeguridad(String usuario);
        
    }
}

File: Administrativo/spring-erp-web/SpringNg/src/app/erp_module/shared/companyowner/servicio/CompanyownerServicio.ts
import { Component, Inject, Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Header } from 'primeng/primeng';
import { DtoEmpleadoBasico } from '../../selectores/empleado/dominio/DtoEmpleadoBasico';
import { DtoTabla } from '../../dominio/dto/DtoTabla';

@Injectable()
export class CompanyownerServicio {

    private rutaServicio: string;
    private http: HttpClient;
    private baseUrl: string;

    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
        this.rutaServicio = this.baseUrl + 'api/spring/core/Companyowner/';
    }

    public listarActivos(): Promise<DtoTabla[]> {
        return this.http.get(this.rutaServicio + 'listarActivos')
            .toPromise()
            .then(response => {
                return response as DtoTabla[];
            })
            .catch(error => new Array());
    }

    public listarCompaniasPorSeguridad(): Promise<DtoTabla[]> {
        return this.http.get(this.rutaServicio + 'listarCompaniasPorSeguridad')
            .toPromise()
            .then(response => {
                return response as DtoTabla[];
            })
            .catch(error => new Array());
    }
}
